### PR TITLE
Enable restoring lost edited content

### PR DESCRIPTION
**Pull request by dawehner**
Follow up for https://github.com/jsdrupal/drupal-admin-ui/pull/357

resolves https://github.com/jsdrupal/drupal-admin-ui/issues/244

### CODE REVIEW

File: src/components/05_pages/NodeEditForm/index.js [hunk 1]
```diff
@@ -77,12 +81,34 @@ styles = {
   `,
 };
 
+const extractRestorableEntity = (state, entity) => {
+  const restorableEntity = state.content.restorableContentEditById[entity.id];
+  if (
+    // @todo maybe that's too defensive?
```
Comment (lauriii) [inline src/components/05_pages/NodeEditForm/index.js]: I think this looks fine to me. I'm wondering if we should have some kind of helper function for this type of conditions.

File: src/components/05_pages/NodeEditForm/index.js [hunk 2]
```diff
@@ -77,12 +81,34 @@ styles = {
...
+    restorableEntity &&
+    restorableEntity.data &&
+    restorableEntity.data.attributes &&
+    entity &&
+    entity.attributes &&
+    // Just offer to restoring when the loaded entity wasn't changed in the meantime.
```
Comment (lauriii) [inline src/components/05_pages/NodeEditForm/index.js]: Maybe `Restoring content is only offered when the loaded entity wasn't changed in the meantime.` would read a bit better.

File: src/reducers/content.js [hunk 3]
```diff
@@ -27,15 +29,23 @@ export default (state = initialState, action) => {
...
-          ...state.contentAddByBundle,
+        restorableContentAddByBundle: {
+          ...state.restorableContentAddByBundle,
           [action.payload.bundle]: action.payload.entity,
         },
       };
+    case CONTENT_EDIT_CHANGE:
+      return {
+        ...state,
+        restorableContentEditById: {
```
Comment (dawehner) [inline src/reducers/content.js]: I'm wondering whether storing every editing content over time is the right approach. Maybe we should limit the amount of stored entires to let's say the last three?

Comment (lauriii) [reply src/reducers/content.js]: Let's open an issue and figure out the correct logic for that. I think for now this is perfectly fine.

Comment (dawehner) [reply src/reducers/content.js]: 👍 Let's be pragmatic indeed

Comment (dawehner) [reply src/reducers/content.js]: Here is one: https://github.com/jsdrupal/drupal-admin-ui/issues/432

File: src/components/05_pages/NodeEditForm/index.js (hunk 2 above)
Comment (dawehner) [reply src/components/05_pages/NodeEditForm/index.js]: That's indeed true. Thank you!

File: src/components/05_pages/NodeEditForm/index.js (hunk 1 above)
Comment (dawehner) [reply src/components/05_pages/NodeEditForm/index.js]: Yeah I guess we could have one of those. The same probably for setters.

**Pull request merged**